internal/blockchain/validate.go

```diff
@@ -3900,20 +3922,34 @@ func (b *BlockChain) checkConnectBlock(node *blockNode, block, parent *dcrutil.B
 		return err
 	}
 
-	const stakeTreeTrue = true
-	err = b.checkTransactionsAndConnect(0, node, block.STransactions(),
-		view, stxos, stakeTreeTrue)
+	// Determine which subsidy split variant to use depending on the agendas
+	// that are active as of the block being checked.
+	isSubsidySplitEnabled, err := b.isSubsidySplitAgendaActive(node.parent)
+	if err != nil {
+		return err
+	}
+	isSubsidySplitR2Enabled, err := b.isSubsidySplitR2AgendaActive(node.parent)
 	if err != nil {
-		log.Tracef("checkTransactionsAndConnect failed for stake tree: %v", err)
 		return err
 	}
+	subsidySplitVariant := standalone.SSVOriginal
+	switch {
+	case isSubsidySplitR2Enabled:
+		subsidySplitVariant = standalone.SSVDCP0012
+	case isSubsidySplitEnabled:
+		subsidySplitVariant = standalone.SSVDCP0010
+	}
```

Comment:
I think you're asking if DCP0010 has to be active in order for DCP0012 to activate?  No, there is no requirement on DCP0010 having been active.  In fact, with newer versions of the test network and the simulation network, it will be the case that they start with DCP0012 from the beginning without DCP0010 ever having been active.

Reply:
It doesn't check both.  Switch statements in Go do not fall through by default (unlike C).  It needs the explicit `fallthrough` keyword to have that behavior.

```Go
package main

import "fmt"

func main() {
	var a, b bool
	setA, setB := true, true
	switch {
	case setA:
		a = true
	case setB:
		b = true
	}
	fmt.Printf("a: %v, b: %v\n", a, b)
}
```

Output:
```
a: true, b: false
```

Reply:
In fact, if it did fall through and check both by default as you're suggesting, the code would be wrong because it would incorrectly end up with the variant set to `standalone.SSVDCP0010` when both agendas are active (which they will be on the main network, because once an agenda is active, it is always active).

Reply:
I mean on line 3927. The values are already checked once they get to the switch. I must be missing something though.

Reply:
The section could be:
```
        var subsidySplitVariant standalone.SubsidySplitVariant
	isSubsidySplitR2Enabled, err := b.isSubsidySplitR2AgendaActive(node.parent)
	if err != nil {
		return err
	}
        if isSubsidySplitR2Enabled {
                subsidySplitVariant = standalone.SSVDCP0012
        } else {
                isSubsidySplitEnabled, err := b.isSubsidySplitAgendaActive(node.parent)
	        if err != nil {
		        return err
                }
                if isSubsidySplitEnabled {
                        subsidySplitVariant = standalone.SSVDCP0012
                }
	}
```

And so sometimes `b.isSubsidySplitAgendaActive` is not done possibly saving some resources. I am either incorrect or unable to convey this though properly though. Apologies.

Reply:
> I mean on line 3927. The values are already checked once they get to the switch. I'm either missing something or unable to convey my thought properly.

Ah, ok, yeah I see what you mean.  Yeah, I suppose it could combine them in that way.  I figured the code is cleaner and easier to follow with it this way and the active checks are realistically just super fast cache lookups in practice, so there isn't really a big performance hit.

e.g. It would otherwise look something like:

```Go
	subsidySplitVariant := standalone.SSVOriginal
	isSubsidySplitR2Enabled, err := b.isSubsidySplitR2AgendaActive(node.parent)
	if err != nil {
		return err
	}
	if !isSubsidySplitR2Enabled {
		isSubsidySplitEnabled, err := b.isSubsidySplitAgendaActive(node.parent)
		if err != nil {
			return err
		}
		if isSubsidySplitEnabled {
			subsidySplitVariant = standalone.SSVDCP0010
		}
	} else {
		subsidySplitVariant = standalone.SSVDCP0012
	}
```

I personally find that harder to read and there is essentially no performance benefit due to the aforementioned cache lookup.

Reply:
Ah, glad I'm not totally crazy. I'm sure it wouldn't be a huge benefit, just a suggestion. Anyway, good job on this, of course.